### ISSUE TITLE
8.0.0 Server Startup won't get past Start Web Server because of Swashbuckle Error

### ISSUE THREAD

**Issue by rbrangers** [StrangeLoopGames/EcoIssues#11128]
[12:00:41] Eco Server 0.8.0.0 beta
[12:00:41] Server Initialization...
[12:00:41] Starting LocalizationPlugin... Finished in 493.5ms
[12:00:42] Starting ModKitPlugin... Finished in 3,356.6ms
[12:00:42] Loading mods...
[12:00:42] Compiling mods... Finished in 2,986.2ms
[12:00:45] Loading mods finished in 3,353.3ms
[12:00:45] Starting NetworkManager... Finished in 64.9ms
[12:00:45] Initializing Simulation Types... Finished in 2,003.8ms
[12:00:45] Starting BackupPlugin... Finished in 8.9ms
[12:00:45] Starting StorageManager... Finished in 3,622.9ms
[12:00:45] Starting WorldGeneratorPlugin... Finished in 406.1ms
[12:00:45] Starting GameplayStatsPlugin... Finished in 0.3ms
[12:00:47] Initializing items... Finished in 4,355.7ms
[12:00:49] Starting PausePlugin... Finished in 37.9ms
[12:00:49] Loading Pause... Finished in 8.5ms
[12:00:51] Initializing skills... Finished in 113.3ms
[12:00:51] Starting StatDatabase... Finished in 21.7ms
[12:00:51] Starting WorldPlugin... Finished in 3,367.9ms
[12:00:52] Loading World... Finished in 2,808.7ms
[12:00:54] Initializing world... Finished in 501.4ms
[12:00:55] Starting TimePlugin... Finished in 2.5ms
[12:00:55] Loading Time... Finished in 1.1ms
[12:00:55] Starting ReputationPlugin... Finished in 6.0ms
[12:00:55] Loading Reputation... Finished in 4.4ms
[12:00:55] Starting PlayerHousingManager... Finished in 1.7ms
[12:00:55] Starting UserManager... Finished in 99.6ms
[12:00:55] Initializing inventories... Finished in 46.2ms
[12:00:55] Starting EcoObjectManager... Finished in 1.4ms
[12:00:55] Starting GarbagePlugin... Finished in 26.8ms
[12:00:55] Starting ControllerManager... Finished in 1,061.2ms
[12:00:55] Starting DifficultySettings... Finished in 13.3ms
[12:00:55] Loading Garbage... Finished in 6.0ms
[12:00:55] Starting ChatServer... Finished in 146.6ms
[12:00:55] Starting EconomyPlugin... Finished in 134.0ms
[12:00:55] Starting DisasterPlugin... Finished in 133.6ms
[12:00:55] Loading Economy... Finished in 126.0ms
[12:00:55] Loading Chat... Finished in 11.1ms
[12:00:55] Starting Legislation... Finished in 148.4ms
[12:00:55] Loading Laws... Finished in 136.0ms
[12:00:55] Starting PropertyManager... Finished in 51.4ms
[12:00:55] Loading Property... Finished in 49.3ms
[12:00:55] Starting MinimapManager... Finished in 5.4ms
[12:00:55] Starting EcoSim... Finished in 1,542.5ms
[12:00:55] Loading Plants... Finished in 697.4ms
[12:00:56] Initializing plants... Finished in 702.5ms
[12:00:57] Starting RouteProbingPlugin... Finished in 551.6ms
[12:00:57] Loading RouteProbing... Finished in 545.7ms
[12:00:57] Starting WorldLayerManager... Finished in 1,461.9ms
[12:00:57] Loading Layers... Finished in 180.5ms
[12:00:58] Initializing world layers... Finished in 1,254.7ms
[12:00:59] Starting EconomyTracker... Finished in 6.7ms
[12:00:59] Starting UserTracker... Finished in 1.8ms
[12:00:59] Starting WorldObjectManager... Finished in 88.4ms
[12:00:59] Loading WorldObjects... Finished in 22.9ms
[12:00:59] Starting RoomPlugin... Finished in 16.1ms
[12:00:59] Loading Rooms... Finished in 12.2ms
[12:00:59] Starting ObjectivePlugin... Finished in 13.8ms
[12:00:59] Loading Objectives... Finished in 1.3ms
[12:00:59] Starting SleepPlugin... Finished in 5.9ms
[12:00:59] Loading Sleep... Finished in 1.0ms
[12:00:59] Starting WebServerPlugin... Finished in 0.2ms
[12:00:59] Starting OtherAliasCategory... Finished in 0.6ms
[12:00:59] Initializing LocalizationPlugin... Finished in 85.6ms
[12:00:59] Initializing ModKitPlugin... Finished in 8.8ms
[12:00:59] Initializing WorldGeneratorPlugin... Finished in 0.5ms
[12:00:59] Initializing UserManager... Finished in 0.5ms
[12:00:59] Initializing GarbagePlugin... Finished in 9.4ms
[12:00:59] Initializing EconomyPlugin... Finished in 11.3ms
[12:00:59] Initializing Legislation... Finished in 560.6ms
[12:01:00] Initializing PropertyManager... Finished in 7.6ms
[12:01:00] Initializing MinimapManager... Finished in 627.7ms
[12:01:00] Initializing UserTracker... Finished in 4.0ms
[12:01:00] Initializing WorldObjectManager... Finished in 99.0ms
[12:01:00] Initializing RoomPlugin... Finished in 1.3ms
[12:01:00] Initializing SleepPlugin... Finished in 0.5ms
[12:01:00] Initializing WorldPlugin... Finished in 562.3ms
[12:01:00] Ticking... Finished in 3.9ms
[12:01:00] Compressing... Finished in 258.6ms
[12:01:01] Initializing NetworkManager... Finished in 39.8ms
[12:01:01] Initializing WorldLayerManager... Finished in 0.4ms
[12:01:01] Initializing WebServerPlugin... Finished in 764.1ms
[12:01:02] Failed to start webserver: Method not found: 'Swashbuckle.Application.SwaggerEnabledConfiguration Swashbuckle.Application.HttpConfigurationExtensions.EnableSwagger(System.Web.Http.HttpConfiguration, System.Action`1<Swashbuckle.Application.SwaggerDocsConfig>)'.
[12:01:02] Server Initialization finished in 20,775.1ms

I removed the entire folder
Downloaded the new clean server and client
Launched the client
Created a New Game
Once the world loaded I did /openserverUI
I made a change in ever tab that I could.
I made a change
Clicked Save
I did this through every tab.
I then copied all the config files from the client's Server folder to the Eco Server folder
Launched the server and still got this error.

This is running on a system with:
Windows 10 Pro
AMD 8-Core processor
24 GB RAM
On a Dedicated Hard drive
 7.8.9 worked with no problems at all

I am also a Full Stack .Net Developer so I have the latest of .Net Framework, .Net Core, IIS, SQL Server.

This is the first time I have every had a problem getting an Eco Server to run.
If you need any more information please feel free to contact me.

**Comment by elfl0rd** [comment]
heh, looks like my old issue - https://github.com/StrangeLoopGames/EcoIssues/issues/10669

**Comment by trueslider** [comment]
Same error on Ubuntu 14.04 Server

[02:03:03] Initializing LocalizationPlugin... ршено за 0,2мс
[02:03:03] Initializing ModKitPlugin...       Завершено за 0,8мс
[02:03:03] Initializing WorldGeneratorPlugin... о за 5,5мс
[02:03:03] Initializing UserManager...          Завершено за 0,2мс
[02:03:03] Initializing GarbagePlugin... вершено за 18,5мс
[02:03:03] Initializing EconomyPlugin... Завершено за 1,7мс
[02:03:03] Initializing Legislation...   Завершено за 6,4мс
[02:03:04] Initializing PropertyManager... ршено за 345,0мс
[02:03:04] Initializing MinimapManager...  Завершено за 2,9мс
[02:03:04] Initializing UserTracker...    Завершено за 712,5мс
[02:03:04] Initializing WorldObjectManager... но за 4,4мс
[02:03:04] Initializing RoomPlugin...         Завершено за 38,8мс
[02:03:04] Initializing SleepPlugin... авершено за 0,6мс
[02:03:04] Initializing WorldPlugin... Завершено за 0,2мс
[02:03:04] Обработка...
[02:03:09] Сжатие...    Завершено за 4 625,1мс
[02:03:09] Initializing NetworkManager... ершено за 4 983,0мс
[02:03:09] Initializing WorldLayerManager... ершено за 128,4мс
[02:03:09] Initializing WebServerPlugin...   Завершено за 0,1мс
[02:03:09] Настройка брандмауэра Windows пропущена, платформа = Unix
[02:03:09] (You may need to manually open port 3001 if the EcoWebServer is inaccessible)
**[02:03:10] Ошибка при запуске веб-сервера: Swashbuckle.Application.SwaggerEnabledConfiguration Swashbuckle.Application.HttpConfigurationExtensions.EnableSwagger(System.Web.Http.HttpConfiguration,System.Action`1<Swashbuckle.Application.SwaggerDocsConfig>)**
[02:03:10] Server Initialization завершена за 54 857,1мс183,7мс
[02:03:10] Настройка брандмауэра Windows пропущена, платформа = Unix
[02:03:10] (You may need to manually open port 3000 if the EcoGameServer is inaccessible)

**Comment by redwyre** [comment]
Can either of you add a log file? There should be a Logs folder under the server directory

**Comment by trueslider** [comment]
Didnt see Logs folder

https://yadi.sk/i/oT601NdR--9BNg

**Comment by kanimaru** [comment]
Same Problem with Ubuntu 18.04, Steam Build, Mono JIT compiler version 4.6.2 (Debian 4.6.2.7+dfsg-1ubuntu1). 

Maybe a custom library missing to pack in build process?

Have also no Log folder.

**Comment by redwyre** [comment]
Sorry, logging isn't working atm, after 8.0.1 releases in the next few days I'll get you guys to try again.

**Comment by gjsmo** [comment]
Encountering the same issue here. I don't have a log folder but I was able to find this in the `Dump` folder: [ServerCrash AggregateException 02094347.ecodmp](https://github.com/StrangeLoopGames/EcoIssues/files/2848116/ServerCrash.AggregateException.02094347.ecodmp.txt)

Searching the error in this appears to indicate that it's related to UPnP being blocked, which it will be on this server, intentionally. I have UPnP disabled in `Network.eco` and both ports appropriately forwarded. Not sure if this helps or if the full log will shed any more light on it but when 8.0.1 releases I can add a log for that as well.

**Comment by Nico640** [comment]
Hi, I have the same problem. I just tried it with the 8.0.1 release. Now it's giving out alot more details.
I've attached the log file.

https://pastebin.com/470WtcvL

**Comment by redwyre** [comment]
@gjsmo I don't think that crash is related, but can you try with 8.0.1 now and attach the log?

Is everyone here using linux?

**Comment by gjsmo** [comment]
I've just upgraded to 8.0.1, same issue. It does not appear to put the same file in `Dump` so it's possible this was a transient issue - I had assumed it was related since it appears to be another network issue.

Anyways, here's the most recent log: [log_190211110247.log](https://github.com/StrangeLoopGames/EcoIssues/files/2858146/log_190211110247.log)

**Comment by Reyalsorik** [comment]
Replicated with a fresh install of 8.0.2 downloaded via Steamcmd. Ran into this issue on Ubuntu 18.04.

Last log before finding this github issue: [log_190214090645.log](https://github.com/StrangeLoopGames/EcoIssues/files/2864204/log_190214090645.log)

**Comment by rbrangers** [comment]
I have an update on this.  This issue is something with the computer not with ECO.  I put a copy on my web server I run at home and it loaded up with no errors while it was throwing this error running it from my gaming rig.  I wiped my gaming rig (Windows 10 Pro System Reset) then re-installed everything and this error is now gone.  I have confirmed this on 8.0.0, 8.0.1, & 8.0.2 on my machines.

**Comment by Artaazar** [comment]
Same here.
Windows Server 2012

[21:31:58.101] [1] [ Info] [Eco] Ticking...  
[21:31:58.101] [1] [ Info] [Eco] Compressing...  
[21:31:58.602] [1] [ Info] [Eco] Initializing NetworkManager...  
[21:31:58.638] [1] [ Info] [Eco] Initializing WorldLayerManager...  
[21:31:58.638] [1] [ Info] [Eco] Initializing WebServerPlugin...  
[21:31:59.472] [1] [ Info] [Eco] Failed to start webserver (TargetInvocationException): Адресат вызова создал исключение.
   в System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   в System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   в System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   в Owin.Loader.DefaultLoader.<>c__DisplayClass12.<MakeDelegate>b__b(IAppBuilder builder)
   в Owin.Loader.DefaultLoader.<>c__DisplayClass1.<LoadImplementation>b__0(IAppBuilder builder)
   в Microsoft.Owin.Hosting.Engine.HostingEngine.Start(StartContext context)
   в Microsoft.Owin.Hosting.Starter.HostingStarter.Start(StartOptions options)
   в Eco.Webserver.WebServer.Init(Int32 port)
Метод не найден: "Swashbuckle.Application.SwaggerEnabledConfiguration Swashbuckle.Application.HttpConfigurationExtensions.EnableSwagger(System.Web.Http.HttpConfiguration, System.Action`1<Swashbuckle.Application.SwaggerDocsConfig>)".
   в Eco.Webserver.Web.Startup.ConfigureWebApi(IAppBuilder appBuilder)
   в Eco.Webserver.Web.Startup.Configuration(IAppBuilder appBuilder)
 
[21:31:59.472] [1] [ Info] [Eco] Initializing IntegrationTestsPlugin...  
[21:31:59.520] [1] [ Info] [Eco] Server Initialization finished in 14 558,3ms

**Comment by gjsmo** [comment]
Another update, I've moved my world to a completely different setup (different network, different hardware, different hypervisor) and get exactly the same issue. Both are running Ubuntu 18.04 LTS server with Mono JIT compiler version 4.6.2 (Debian 4.6.2.7+dfsg-1ubuntu1).

**Comment by Koden02** [comment]
Same here, Windows 10
Eco Server 0.8.0.3 beta


[14:03:33.428] [1] [ Info] [Eco] Ticking...  
[14:03:36.360] [1] [ Info] [Eco] Compressing...  
[14:03:37.435] [1] [ Info] [Eco] Initializing NetworkManager...  
[14:03:37.484] [1] [ Info] [Eco] Initializing WorldLayerManager...  
[14:03:37.484] [1] [ Info] [Eco] Initializing WebServerPlugin...  
[14:03:38.236] [1] [ Info] [Eco] Failed to start webserver (TargetInvocationException): Exception has been thrown by the target of an invocation.
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Owin.Loader.DefaultLoader.<>c__DisplayClass12.<MakeDelegate>b__b(IAppBuilder builder)
   at Owin.Loader.DefaultLoader.<>c__DisplayClass1.<LoadImplementation>b__0(IAppBuilder builder)
   at Microsoft.Owin.Hosting.Engine.HostingEngine.Start(StartContext context)
   at Microsoft.Owin.Hosting.Starter.HostingStarter.Start(StartOptions options)
   at Eco.Webserver.WebServer.Init(Int32 port)
Method not found: 'Swashbuckle.Application.SwaggerEnabledConfiguration Swashbuckle.Application.HttpConfigurationExtensions.EnableSwagger(System.Web.Http.HttpConfiguration, System.Action`1<Swashbuckle.Application.SwaggerDocsConfig>)'.
   at Eco.Webserver.Web.Startup.ConfigureWebApi(IAppBuilder appBuilder)
   at Eco.Webserver.Web.Startup.Configuration(IAppBuilder appBuilder)
 
[14:03:38.236] [1] [ Info] [Eco] Initializing IntegrationTestsPlugin...  
[14:03:38.281] [1] [ Info] [Eco] Server Initialization finished in 21,537.3ms

**Comment by Koden02** [comment]
I'm also running the server as admin.

**Comment by SagaciousZed** [comment]
Running Single Player world on  Windows 10 Pro 1083

```
[10:39:52.382] [1] [ Info] [Eco] Failed to start webserver (TargetInvocationException): Exception has been thrown by the target of an invocation.
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Owin.Loader.DefaultLoader.<>c__DisplayClass12.<MakeDelegate>b__b(IAppBuilder builder)
   at Owin.Loader.DefaultLoader.<>c__DisplayClass1.<LoadImplementation>b__0(IAppBuilder builder)
   at Microsoft.Owin.Hosting.Engine.HostingEngine.Start(StartContext context)
   at Microsoft.Owin.Hosting.Starter.HostingStarter.Start(StartOptions options)
   at Eco.Webserver.WebServer.Init(Int32 port)
Method not found: 'Swashbuckle.Application.SwaggerEnabledConfiguration Swashbuckle.Application.HttpConfigurationExtensions.EnableSwagger(System.Web.Http.HttpConfiguration, System.Action`1<Swashbuckle.Application.SwaggerDocsConfig>)'.
   at Eco.Webserver.Web.Startup.ConfigureWebApi(IAppBuilder appBuilder)
   at Eco.Webserver.Web.Startup.Configuration(IAppBuilder appBuilder)
 
```

I've tried tweaking some optional Windows features for IIS, but I am unable to find a working combination, if it is indeed related at all.

**Comment by Cherry** [comment]
Hey all. Following up here as we appear to have found a fix for the SwashBuckle issue (at least on Linux, game version 0.8.0.3).

After discovering https://hub.docker.com/r/strangeloopgames/eco-game-server worked fine, we examined the differences and found that if you include a specific [`EcoServer.exe.config`](https://backups-b2.nodecraft.gg/file/nodecraft-assets/articles/eco/EcoServer.exe.config) file in the root directory of your server, the error disappears and the webserver appears to work without issues.

We've published a short article with more information at https://nodecraft.com/support/games/eco/fixing-the-eco-webserver-swashbuckle-error

**Comment by Artaazar** [comment]
[image: image.png]
[image: image.png]

there is no .config file in any version of server

пт, 22 февр. 2019 г. в 01:14, James Ross <notifications@github.com>:

**Comment by Cherry** [comment]
> [image: image.png] [image: image.png] there is no .config file in any version of server 

Not by default, that's correct. Hopefully the Eco team can begin shipping this with the server files in a future update, but for now, you need to add it to your server via the instructions I posted above. 😃

**Comment by Koden02** [comment]
> 
> 
> Hey all. Following up here as we appear to have found a fix for the Swashbuckle issue (at least on Linux, game version 0.8.0.3).
> 
> After discovering https://hub.docker.com/r/strangeloopgames/eco-game-server worked fine, we examined the differences and found that if you include a specific [`EcoServer.exe.config`](https://backups-b2.nodecraft.gg/file/nodecraft-assets/articles/eco/EcoServer.exe.config) file in the root directory of your server, the error disappears and the webserver appears to work without issues.
> 
> We've published a short article with more information at https://nodecraft.com/support/games/eco/fixing-the-eco-webserver-swashbuckle-error

I'm on windows 10 and this fixed my issue.

**Comment by Artaazar** [comment]
It was just for information.
But, prev version (7.5.*) still have no .config file, but web server works
fine :)
[image: image.png]1
пт, 22 февр. 2019 г. в 02:52, James Ross <notifications@github.com>:

**Comment by Rambo99** [comment]
Steam game version 0.8.0.3b . 
[09:30:52.660] [1] [ Info] [Eco] Failed to start webserver (TargetInvocationException): Адресат вызова создал исключение.
   в System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   в System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   в System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   в Owin.Loader.DefaultLoader.<>c__DisplayClass12.<MakeDelegate>b__b(IAppBuilder builder)
   в Owin.Loader.DefaultLoader.<>c__DisplayClass1.<LoadImplementation>b__0(IAppBuilder builder)
   в Microsoft.Owin.Hosting.Engine.HostingEngine.Start(StartContext context)
   в Microsoft.Owin.Hosting.Starter.HostingStarter.Start(StartOptions options)
   в Eco.Webserver.WebServer.Init(Int32 port)
Метод не найден: "Swashbuckle.Application.SwaggerEnabledConfiguration Swashbuckle.Application.HttpConfigurationExtensions.EnableSwagger(System.Web.Http.HttpConfiguration, System.Action`1<Swashbuckle.Application.SwaggerDocsConfig>)".
   в Eco.Webserver.Web.Startup.ConfigureWebApi(IAppBuilder appBuilder)
   в Eco.Webserver.Web.Startup.Configuration(IAppBuilder appBuilder)

**Comment by redwyre** [comment]
This fix is in 0.8.0.4 which should be released in the next few days.
